efs/efs.tf

```diff
@@ -4,13 +4,33 @@ resource "aws_efs_file_system" "project_efs" {
   creation_token = "${var.project_name}-file-system"
   throughput_mode = "bursting"
   performance_mode = "generalPurpose"
-  encrypted = false
+  encrypted = true
 }
 
-# EFS Moutn Target
+# EFS Mount Target
 
 resource "aws_efs_mount_target" "project_mount_target" {
   file_system_id = aws_efs_file_system.project_efs.id
   subnet_id      = var.private_subnet_id
-  security_groups = [var.security_groups[0].id]
+  security_groups = [aws_security_group.project_security_group.id]
+}
+
+# Security Group
+
+resource "aws_security_group" "project_security_group" {
+  name        = var.sg_name
+  description = "Security Group for Elastic File System"
+  vpc_id      = var.vpc_id
+
+ingress {
+    description = "ECS Access"
+    from_port   = 2049
+    to_port     = 2049
+    protocol    = "tcp"
+    cidr_blocks = [var.vpc_cidr]
```

Comment:
need to use the security_group_rule to be able to use a security_group_id as a source instead of a CIDR block

Reply:
I used the attribute`security_groups` from aws_security_group and it seems to work